R/g.report.part5.R

```diff
@@ -9,42 +10,61 @@ g.report.part5 = function(metadatadir = c(), f0 = c(), f1 = c(), loglocation = c
+      } else if (includedaycrit.part5 < 0 ) {
+        warning("\nNegative value of includedaycrit.part5 is not allowed, please change.")
+      } else if (includedaycrit.part5 > 25) {
+        warning("\nIncorrect value of includedaycrit.part5, this should be a fraction of the day between zero and one or the number of hours in a day.")
+      }
+      maxpernwday = 100 - includedaycrit.part5
+      include_window = rep(TRUE, nrow(x))
+      if (length(data_cleaning_file) > 0) { # allow for forced relying on guider based on external data_cleaning_file
+        DaCleanFile = data.table::fread(data_cleaning_file, data.table = FALSE)
+        days2exclude = which(DaCleanFile$ID %in% x$ID & DaCleanFile$day_part5 %in% x$window_number)
+        if (length(days2exclude) > 0) {
+          for (ri in 1:length(days2exclude)) {
+            id2remove = DaCleanFile$ID[days2exclude[ri]]
+            window2remove = DaCleanFile$day_part5[days2exclude[ri]]
+            include_window[which(x$ID == id2remove & x$window_number == window2remove)] = FALSE
+          }
         }
+      } else {
+        include_window = rep(TRUE,nrow(x))
       }
-    } else {
-      include_window = rep(TRUE,nrow(x))
-    }
-    # Note: Below we intentionally only sets a criteria on daytime, because for
-    # the night time we only need start and end of the SPT window.
-    if (window == "WW") {
-      indices = which(x$nonwear_perc_day <= maxpernwday &
-                        x$dur_spt_min > 0 & x$dur_day_min > 0 & include_window == TRUE)
-    } else if (window == "MM") {
-      indices = which(x$nonwear_perc_day <= maxpernwday &
-                        x$dur_spt_min > 0 & x$dur_day_min > 0 &
-                        x$dur_day_spt_min >= (minimum_MM_length.part5*60) &
+      # Note: Below we intentionally only sets a criteria on daytime, because for
+      # the night time we only need start and end of the SPT window.
+      if (window == "WW") {
+        indices = which(x$nonwear_perc_day <= maxpernwday &
+                          x$dur_spt_min > 0 & x$dur_day_min > 0 & include_window == TRUE)
+      } else if (window == "MM") {
+        indices = which(x$nonwear_perc_day <= maxpernwday &
+                          x$dur_spt_min > 0 & x$dur_day_min > 0 &
+                          x$dur_day_spt_min >= (minimum_MM_length.part5*60) &
+                          include_window == TRUE)
+        # Note: By default for MM analysis only full days are interesting (23 hours for one day in the year)
+      }
+    } else if (window == "Segments") {
+      # clean based on segments (even if a day is not valid, a certain segment of that
+      # day could be valid if participant wore device enough in that part of the day)
+      if (segmentWEARcrit.part5 >= 0 & segmentWEARcrit.part5 <= 1) { # if segmentWEARcrit.part5 is used as a ratio
+        segmentWEARcrit.part5 = segmentWEARcrit.part5 * 100
+      } else if (segmentWEARcrit.part5 < 0 ) {
```

Comment:
Can you move these checks on the value of segmentWEARcrit.part5 and segmentWEARcrit.part5 to params_check, in that way we keep all basic checks of input argument there. This is preferable because want to warn user about mistakes before processing all the data.